src/osd/OSD.cc

```diff
@@ -2633,17 +2633,19 @@ int OSD::init()
     }
   }
 
-  r = update_crush_device_class();
-  if (r < 0) {
-    derr << __func__ <<" unable to update_crush_device_class: "
-         << cpp_strerror(r) << dendl;
-    osd_lock.Lock();
-    goto monout;
+  if (osdmap->require_osd_release >= CEPH_RELEASE_LUMINOUS) {
```

Comment:
This seems not to be an ideal fix.  For newly created osds even currently running on a Luminous cluster, the above condition will prevent them from automatically setting device classes as they should still hold empty osdmaps by then...
I'd instead suggest:
(1) simply  ignore the result of update_crush_device_class(), we can still manually 'set-device-class' later;
(2) wait until monitor upgrade is done and then retry (update_crush_device_class())